sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveStatsFunctions.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.expressions.aggregate._
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreePattern.SUMMARY
+import org.apache.spark.sql.types._
+
+/**
+ * Resolve StatsFunctions.
+ */
+object ResolveStatsFunctions extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsWithPruning(
+    _.containsPattern(SUMMARY), ruleId) {
+
+    case s @ UnresolvedSummary(child, statistics) if s.childrenResolved =>
+      val percentiles = statistics.filter(p => p.endsWith("%"))
+        .map(p => p.stripSuffix("%").toDouble / 100.0)
+
+      var mapExprs = Seq.empty[NamedExpression]
+      child.output.foreach { attr =>
+        if (attr.dataType.isInstanceOf[NumericType] || attr.dataType.isInstanceOf[StringType]) {
+          val name = attr.name
+          val casted: Expression = attr.dataType match {
+            case StringType => Cast(attr, DoubleType, evalMode = EvalMode.TRY)
+            case _ => attr
+          }
+
+          val approxPercentile = if (percentiles.nonEmpty) {
+            Alias(
+              new ApproximatePercentile(
+                casted,
+                Literal(percentiles.toArray),
+                Literal(ApproximatePercentile.DEFAULT_PERCENTILE_ACCURACY)
+              ).toAggregateExpression(),
+              s"__${name}_approx_percentile__")()
+          } else null
```

Comment:
I am see a potential issue of NPE. Is there a way that we get rid of `null` here?

Reply:
Yeah looks this `val` is passed to `new Get` and there is no defensive check over `null` there. If there is ever a `null` case that appears, it will hit NPE.